### PR TITLE
Hw11 deeplink

### PR DESCRIPTION
**Pull request by alexey-chaichuk**
Screenshot_1613136281
Screenshot_1613136286
Screenshot_1613136291
Screenshot_1613136295
Screenshot_1613136461
Screenshot_1613136469
Screenshot_1613136478

### CODE REVIEW

File: app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt [hunk 1]
```diff
@@ -17,6 +22,28 @@ class MainActivity :
...
+        intent?.let(::handleIntent)
+    }
+
+    private fun handleIntent(intent: Intent) {
+        when (intent.action) {
+            Intent.ACTION_VIEW -> {
+                val id = intent.data?.lastPathSegment?.toIntOrNull()
+                id?.apply {
+                    Log.d("HWApp", "Handle intent : ${id}")
+                    runBlocking {
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt]: never use runBlocking 
on activity has lifeCycleScope object 
let's create coroutine from it

File: app/src/main/java/ru/chaichuk/hwapp/api_v3/MovieDbApi.kt [hunk 2]
```diff
@@ -58,6 +58,38 @@ class MovieDbApi {
...
+                true
+            )
+        }
+
+        return movie
+    }
+
     private suspend fun getPopularMovies(page : Int) : List <MovieDTO> = withContext(Dispatchers.IO) {
         val moviesPage = RetrofitModule.moviesApi.popularMovies(page = page).log("getPopularMovies api item")
         return@withContext moviesPage.movies
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/api_v3/MovieDbApi.kt]: I think you can use just `moviesPage.movies` withOut return

Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/api_v3/MovieDbApi.kt]: changed function body to
RetrofitModule.moviesApi.popularMovies(page = page).log("getPopularMovies api item").movies
without return

File: app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt (hunk 1 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt]: changed to
lifecycleScope.launch() {

**Review by Axrorxoja — APPROVED** [review]

**Pull request merged**